async-nats/benches/core_nats.rs

```diff
@@ -82,9 +81,12 @@ pub fn subscribe(c: &mut Criterion) {
                             let msg = &bmsg[0..*size].to_vec();
 
                             loop {
-                                nc.publish("bench".to_string(), msg.clone().into())
-                                    .await
-                                    .unwrap();
+                                let result =
```

Comment:
Maybe better directly check for disconnect event on the client and the break the loop, log publish errors (but not panic)?

The benchmark results and report would be cleaner about what happened.

Reply:
So about printing the error, I don't want to give the impression that the benchmark failed when its expected (albeit racy behavior).

Could instead just loop from 0 to N and still unwrap.

Reply:
I get your point and agree with it.
I'm talking about maybe having more graceful and controlled shutdown of server/clients in benchmarks.